aquadoggo/migrations/20220509090252_create-operations.sql

```diff
@@ -0,0 +1,29 @@
+-- SPDX-License-Identifier: AGPL-3.0-or-later
+
+CREATE TABLE IF NOT EXISTS operations_v1 (
+    author              VARCHAR(64)       NOT NULL,
+    document_id         VARCHAR(64)       NOT NULL,
+    operation_id        VARCHAR(64)       NOT NULL UNIQUE,
+    entry_hash          VARCHAR(68)       NOT NULL UNIQUE,
+    action              VARCHAR(16)       NOT NULL,
+    schema_id           TEXT              NOT NULL,
+    previous_operations TEXT              NULL,
+    PRIMARY KEY (operation_id)
+);
+
+-- With the above "previous_operations" column in operations_v1 table we may no longer need this 
+-- relation table. Can we remove it or are there reasons it's useful?
+CREATE TABLE IF NOT EXISTS previous_operations_v1 (
+    parent_operation_id    VARCHAR(64)       NOT NULL,
+    child_operation_id     VARCHAR(64)       NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS operation_fields_v1 (
+    operation_id           VARCHAR(64)       NOT NULL,
+    name                   VARCHAR(128)      NOT NULL,
+    field_type             TEXT              NOT NULL,
+    value                  BLOB              NULL,
```

Comment:
I'd like to somehow make value a typed field, I think this could be achieved by:
- having a different table for everuy field type
- having a column for each field type in the fields table and only populating the relevant one (it's possible to restrict the columns in a "only one is not null" way).

Reply:
But this could happen in a later PR, we still aren't 100% committed to this table layout I don't think.